Assets/Tests/PlayModeTests/ConsentTests.cs

```diff
@@ -0,0 +1,507 @@
+        }
+
+        [Test]
+        public void TestListenerOnConsentChanged()
+        {
+            CountlyConfiguration configuration = new CountlyConfiguration {
+                AppKey = _appKey,
+                ServerUrl = _serverUrl,
+                RequiresConsent = true,
+            };
+
+
+            ConsentTestHelperClass listener = new ConsentTestHelperClass();
+            ConsentCountlyService consentCountlyService = new ConsentCountlyService(configuration, null);
+
+            consentCountlyService.Listeners = new List<AbstractBaseService> { listener };
+
+            consentCountlyService.GiveConsent(new Consents[] { Consents.Location, Consents.RemoteConfig, Consents.Events });
+            Assert.AreEqual(true, listener.Validate(0, new Consents[] { Consents.Location, Consents.RemoteConfig, Consents.Events }, true));
+
+            consentCountlyService.GiveConsent(new Consents[] { Consents.Location, Consents.StarRating });
+            Assert.AreEqual(1, listener.DeltaConsentsList[1].updatedConsents.Count);
+            Assert.AreEqual(true, listener.Validate(1, new Consents[] { Consents.StarRating }, true));
+
+
+            consentCountlyService.RemoveConsent(new Consents[] { Consents.Location, Consents.StarRating });
+            Assert.AreEqual(2, listener.DeltaConsentsList[2].updatedConsents.Count);
+            Assert.AreEqual(true, listener.Validate(2, new Consents[] { Consents.Location, Consents.StarRating }, false));
+
+
+            consentCountlyService.RemoveConsent(new Consents[] { Consents.Events, Consents.StarRating });
+            Assert.AreEqual(1, listener.DeltaConsentsList[3].updatedConsents.Count);
+            Assert.AreEqual(true, listener.Validate(3, new Consents[] { Consents.Events }, false));
+        }
+
+        [Test]
+        public void TestListenerOnConsentGroups()
+        {
+            CountlyConfiguration configuration = new CountlyConfiguration {
+                AppKey = _appKey,
+                ServerUrl = _serverUrl,
+                RequiresConsent = true,
+            };
+
+            string groupA = "GroupA";
+            string groupB = "GroupB";
+
+            configuration.CreateConsentGroup(groupA, new Consents[] { Consents.Clicks, Consents.Views });
+
+            ConsentTestHelperClass listener = new ConsentTestHelperClass();
+            ConsentCountlyService consentCountlyService = new ConsentCountlyService(configuration, null);
+
+            consentCountlyService.Listeners = new List<AbstractBaseService> { listener };
+
+            consentCountlyService.GiveConsentToGroup(new string[] { groupA });
+            Assert.AreEqual(true, listener.Validate(0, new Consents[] { Consents.Clicks, Consents.Views }, true));
+
+            consentCountlyService.GiveConsentToGroup(new string[] { groupB });
+            Assert.AreEqual(false, listener.Validate(1, new Consents[] { Consents.Clicks, Consents.Views }, true));
```

Comment:
replace this check with as assert that the internal listener DeltaConsentsList count has not incresed and is still 1